sympy/solvers/tests/test_recurr.py

```diff
@@ -204,6 +204,16 @@ def test_issue_6844():
     assert rsolve(f, y(n), {y(0): 0, y(1): 1}) == 2*2**(-n)*n
 
 
+def test_issue_18751():
+    n = Symbol('n', integer=True)
+    y = Function('y')
```

Comment:
It is already declared on the top. Maybe we can remove this.

Reply:
Done.